cloud/services/managedclusters/managedclusters.go

```diff
@@ -123,50 +125,36 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 			},
 			AgentPoolProfiles: &[]containerservice.ManagedClusterAgentPoolProfile{},
 			NetworkProfile: &containerservice.NetworkProfileType{
-				NetworkPlugin:   containerservice.Azure,
-				LoadBalancerSku: containerservice.Standard,
+				NetworkPlugin:   containerservice.NetworkPlugin(*managedClusterSpec.NetworkPlugin),
```

Comment:
deref without nil check on managedClusterSpec NetworkPlugin/LoadBalancerSKU/NetworkPolicy. It's passthrough from the control plane which is optional, albeit set in the webhook. 

Could you make these three values non-pointer in the managedClusterSpec used by the service, remove the deref here, and conditionally set them if not null on the managedClusterSpec instead?

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/862/files#diff-78feb4e2f24becff4de1a0bc4348e52bR108-R112

Reply:
incidentally, I think a nil deref on one of these is the cause of the test failure

Reply:
good catch, that is a cardinal sin of pointers :sweat_smile: